[Docs] Updates the SET command documentation (#18101)
apache/druid: docs/querying/sql.md

```
[ EXPLAIN PLAN FOR ]
[ WITH tableName [ ( column1, column2, ... ) ] AS ( query ) ]

> [ SET <parameter> = <value>;]

Author: Just removed from it the section :)

Request logs show the exact native query that will be run. Alternatively, to see the native query plan, set `useNativeQueryExplain` to true in the query context.
:::

## SET 

> `SET` statements allow you to specify SQL query context parameters that modify the behavior of a Druid SQL query. These statements can be included before the main SQL query and are supported in multiple query interfaces, including the The Druid SQL [JSON API](../api-reference/sql-api.md) and the [Druid Web Console](../operations/web-console.md). 

Reviewer: ```suggestion
SET statements allow you to specify SQL query context parameters that modify the behavior of a Druid SQL query. You can include one or more SET statements before the main SQL query. Druid supports using SET in the Druid SQL [JSON API](../api-reference/sql-api.md) and the [web console](../operations/web-console.md). 
```

Request logs show the exact native query that will be run. Alternatively, to see the native query plan, set `useNativeQueryExplain` to true in the query context.
:::

## SET 

> `SET` statements allow you to specify SQL query context parameters that modify the behavior of a Druid SQL query. These statements can be included before the main SQL query and are supported in multiple query interfaces, including the The Druid SQL [JSON API](../api-reference/sql-api.md) and the Druid Web Console. 

Reviewer: ```suggestion
SET statements allow you to specify SQL query context parameters that modify the behavior of a Druid SQL query. You can include one or more SET statements before the main SQL query. Druid supports using SET in the Druid SQL [JSON API](../api-reference/sql-api.md) and the web console.
```

Reviewer: Also add a link to web console: https://druid.apache.org/docs/latest/operations/web-console

> You can include 0 or more `SET` statements, each separated by a semicolon `;`, preceding a statement to execute in the `query` string of the request. 

Reviewer: This is now covered by information in the preceding and following sections.
```suggestion
```

SELECT some_column, COUNT(*) FROM druid.foo WHERE other_column = 'foo' GROUP BY 1 ORDER BY 2 DESC
```

> If present, these `SET` statements assign [SQL query context parameter](../querying/sql-query-context.md) which only apply to the non-`SET` statement of the same request and subsequent requests are not affected.

Reviewer: ```suggestion
SET statements only apply to the query in the same request. Subsequent requests are not affected.
```

> You can mix `SET` statement and [Context Parameter](../querying/query-context.md) with `SET` overriding on key conflicts.

Reviewer: ```suggestion
If you use the [JSON API](../api-reference/sql-api.md), you can also include query context parameters using the `context` field. If you include both, the parameter value in SET takes precedence over the parameter value in `context`.
```

> ### Current limitations
> 
> - `SET` statements currently support literal values only. For example, numbers, strings, booleans.
> -  You must still provide arrays or JSON objects via the `context` field.
> 
> 

Reviewer: This is really two ways to say the same thing. We don't need to create a new section for just one limitation.

```suggestion
Note that you can only use SET to assign literal values, such as numbers, strings, or Booleans. To set a query context parameter to an array or JSON object, use the `context` field rather than SET.
```
